### PR TITLE
Added additional functions

### PR DESCRIPTION
**Pull request by stephanheilner**

### CODE REVIEW

File: Swiftification/Array.swift [hunk 1]
```diff
@@ -98,5 +98,25 @@ public extension Array {
         
         return allGroups
     }
+        
+    /// Returns an array containing the the last numberOfElements elements of self.
+    @warn_unused_result
+    func tail(numberOfElements: Int) -> Array {
+        return Array(self[(count - numberOfElements)..<count])
```
Comment (hiltonc) [inline Swiftification/Array.swift]: Add test for `[1, 2].tail(10)` or something similar.

File: SwiftificationTests/ArrayTests.swift [hunk 2]
```diff
@@ -84,4 +84,22 @@ class ArrayTests: XCTestCase {
         XCTAssertTrue(sectioned[1].items.count == 3)
     }
     
+    func testTail() {
+        let array = Array(1...10)
+        XCTAssertTrue(array.tail(1) == [10])
+        XCTAssertTrue(array.tail(3) == [8, 9, 10])
+    }
+    
+    func testShift() {
```
Comment (hiltonc) [inline SwiftificationTests/ArrayTests.swift]: Also test an empty array.

File: Swiftification/Array.swift [hunk 3]
```diff
@@ -98,5 +98,25 @@ public extension Array {
...
+    mutating func shift() -> Element? {
+        if !self.isEmpty {
+            return removeAtIndex(0)
+        }
+        return nil
+    }
     
+    /// Returns an array containing the first n elements of self.
+    @warn_unused_result
+    func take(numberOfElements: Int) -> Array {
```
Comment (hiltonc) [inline Swiftification/Array.swift]: Test `[1, 2].take(10)` or similar.

File: Swiftification/SequenceType.swift [hunk 4]
```diff
@@ -46,4 +46,42 @@ public extension SequenceType {
         return nil
     }
     
+    /// Returns the array of elements for which condition(element) is unique
+    @warn_unused_result
+    func uniqueBy<T: Equatable>(condition: (Generator.Element) -> T) -> [Generator.Element] {
+        var results: [Generator.Element] = []
+        var tempResults: [T] = []
```
Comment (hiltonc) [inline Swiftification/SequenceType.swift]: Use a set for `tempResults` for better perf.

File: Swiftification/SequenceType.swift [hunk 5]
```diff
@@ -46,4 +46,42 @@ public extension SequenceType {
...
+   
+}
+
+public extension SequenceType where Generator.Element: Equatable {
+    
+    /// Returns an array removing the duplicate elements in self if Generator implements the Equatable protocol.
+    @warn_unused_result
+    func unique() -> [Generator.Element] {
+        var results: [Generator.Element] = []
+        for element in self where !results.contains(element) {
```
Comment (hiltonc) [inline Swiftification/SequenceType.swift]: Use a temp set for better perf.

File: SwiftificationTests/SequenceTypeTests.swift [hunk 6]
```diff
@@ -39,4 +39,24 @@ class SequenceTypeTests: XCTestCase {
         XCTAssertEqual(expected, actual)
     }
     
-}
+    func testUnique() {
+        let array = [1, 1, 2, 2, 3, 4]
+        let unique = array.unique()
+        XCTAssertTrue(array.count == 6)
+        XCTAssertTrue(unique.count == 4)
```
Comment (nickmshelley) [inline SwiftificationTests/SequenceTypeTests.swift]: I would test that `unique` is equal to `[1, 2, 3, 4]` since I think preserving order is important.

File: SwiftificationTests/SequenceTypeTests.swift [hunk 7]
```diff
@@ -39,4 +39,24 @@ class SequenceTypeTests: XCTestCase {
...
+    }
+    
+    func testUniqueBy() {
+        let array = [(a: 1, b: 1), (a: 1, b: 2), (a: 2, b: 2)]
+        let unique = array.uniqueBy { $0.a }
+        XCTAssertTrue(array.count == 3)
+        XCTAssertTrue(unique.count == 2)
```
Comment (nickmshelley) [inline SwiftificationTests/SequenceTypeTests.swift]: Same as above, I'd check it's equal to `[1, 2]` to verify order.

File: SwiftificationTests/SequenceTypeTests.swift (hunk 6 above)
Comment (stephanheilner) [inline SwiftificationTests/SequenceTypeTests.swift]: On it...

**Comment by hiltonc** [comment]
Can you inc pod version to 2.2.0? Otherwise looks great. Sorry generics are such a beast.
